File: HelpReviews.client/src/models/Restaurant.js
import { Profile } from "./Profile.js"

export class Restaurant {
    constructor(data) {
        this.id = data.id
        this.name = data.name
        this.imgUrl = data.imgUrl
        this.description = data.description
        this.visits = data.visits
        this.isShutdown = data.isShutdown
        this.creatorId = data.creatorId
        this.creator = new Profile(data.creator)
    }
}

File: HelpReviews/Models/Restaurant.cs
namespace HelpReviews.Models;

public class Restaurant
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string ImgUrl { get; set; }
    public string Description { get; set; }
    public int Visits { get; set; }
    public int ReportCount { get; set; }
    public bool? IsShutdown { get; set; }
    public string CreatorId { get; set; }
    public Profile Creator { get; set; }
}